streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -479,6 +486,21 @@ private RuntimeException performOnTasks(final List<AbstractTask> tasks,
         return firstException;
     }
 
+    private void closeAllProducers() {
```

Comment:
Similar to my other comment, it seems that we should be thinking about closing tasks. That way, if we have other resources associated with a task in the future, we don't have to change any code.